Code/Tools/AssetProcessor/native/FileWatcher/FileWatcher.h

```diff
@@ -74,23 +32,41 @@ class FileWatcher
 
 public:
     FileWatcher();
-    virtual ~FileWatcher();
+    ~FileWatcher() override;
 
     //////////////////////////////////////////////////////////////////////////
-    virtual int AddFolderWatch(FolderWatchBase* pFolderWatch, bool recursive = true);
-    virtual void RemoveFolderWatch(int handle);
+    void AddFolderWatch(QString directory, bool recursive = true);
+    void ClearFolderWatches();
     //////////////////////////////////////////////////////////////////////////
-    
+
     void StartWatching();
     void StopWatching();
 
 Q_SIGNALS:
-    void AnyFileChange(FileChangeInfo info);
+    void fileAdded(QString filePath);
+    void fileRemoved(QString filePath);
+    void fileModified(QString filePath);
+
+    void rawFileAdded(QString filePath);
+    void rawFileRemoved(QString filePath);
+    void rawFileModified(QString filePath);
```

Comment:
Can you comment when these signals are fired and what they are used for?

For example it seems that `rawFile*` signals does not run the filePath through a filter.

Reply:
I made the `raw` signals use `QPrivateSignal` and added some comments to clarify how it works.